### PR TITLE
feat: add sql to substrait conversion

### PR DESCRIPTION
**Pull request by EpsilonPrime**
This conversion converts the provided SQL into Substrait.  Since the conversion is done by
the DuckDB routines under the hood the SQL dialect and Substrait dialect are both DuckDB.

### CODE REVIEW

File: src/gateway/converter/spark_to_substrait_test.py [hunk 1]
```diff
@@ -10,20 +10,26 @@
...
-test_case_paths = [f for f in test_case_directory.iterdir() if f.name.endswith('.spark')]
+substrait_test_case_paths = [f for f in test_case_directory.iterdir() if f.name.endswith('.spark')]
+
+substrait_test_case_names = [
+    os.path.basename(p).removesuffix('.spark') for p in substrait_test_case_paths]
 
-test_case_names = [os.path.basename(p).removesuffix('.spark') for p in test_case_paths]
+sql_test_case_paths = [f for f in test_case_directory.iterdir() if f.name.endswith('.sql')]
+
+sql_test_case_names = [os.path.basename(p).removesuffix('.sql') for p in sql_test_case_paths]
```
Comment (richtia) [inline src/gateway/converter/spark_to_substrait_test.py:25]: if p is just a Path here, you can probably do p.stem to remove the extension

File: src/gateway/converter/spark_to_substrait_test.py [hunk 2]
```diff
@@ -10,20 +10,26 @@
 
 from gateway.converter.conversion_options import duck_db
 from gateway.converter.spark_to_substrait import SparkSubstraitConverter
-
+from gateway.converter.sql_to_substrait import SqlConverter
+from gateway.demo.mystream_database import create_mystream_database, delete_mystream_database
 
 test_case_directory = Path(os.path.dirname(os.path.realpath(__file__))) / 'data'
```
Comment (richtia) [inline src/gateway/converter/spark_to_substrait_test.py:16]: would something like this work?
```
Path(__file__).resolve().parent / "data"
```

File: src/gateway/converter/spark_to_substrait_test.py (hunk 1 above)
Comment (EpsilonPrime) [reply src/gateway/converter/spark_to_substrait_test.py]: Done.

File: src/gateway/converter/spark_to_substrait_test.py (hunk 2 above)
Comment (EpsilonPrime) [reply src/gateway/converter/spark_to_substrait_test.py]: Done, much cleaner now.

**Pull request merged**